addons/giltracker/giltracker.lua

```diff
@@ -0,0 +1,182 @@
+defaults.GilText.text.red = 253
+defaults.GilText.text.green = 252
+defaults.GilText.text.blue = 250
+defaults.GilText.text.stroke = {}
+defaults.GilText.text.stroke.alpha = 200
+defaults.GilText.text.stroke.red = 50
+defaults.GilText.text.stroke.green = 50
+defaults.GilText.text.stroke.blue = 50
+defaults.GilText.text.stroke.width = 2
+defaults.GilText.text.visible = true
+defaults.GilImage = {}
+defaults.GilImage.color = {}
+defaults.GilImage.color.alpha = 255
+defaults.GilImage.color.red = 255
+defaults.GilImage.color.green = 255
+defaults.GilImage.color.blue = 255
+defaults.GilImage.visible = true
+
+local settings = config.load(defaults)
+config.save(settings)
+
+settings.GilImage.texture = {}
+settings.GilImage.texture.path = windower.addon_path..'gil.png'
+settings.GilImage.texture.fit = true
+settings.GilImage.size = {}
+settings.GilImage.size.height = 23
+settings.GilImage.size.width = 23
+settings.GilImage.draggable = false
+settings.GilImage.repeatable = {}
+settings.GilImage.repeatable.x = 1
+settings.GilImage.repeatable.y = 1
+
+gil_image = images.new(settings.GilImage)
+gil_text = texts.new(settings.GilText)
+
+windower.register_event('load',function()
+    if windower.ffxi.get_info().logged_in then
+        initialize()
+    end
+end)
+
+windower.register_event('login', function()
+    initialize()
+end)
+
+windower.register_event('logout', function(...)
+    hide()
+end)
+
+windower.register_event('add item', function(...)
+    update_gil()
+end)
+
+windower.register_event('remove item', function(original, modified, original_mode, modified_mode, blocked)
+    if (string.match(original,"gil")) then
+        update_gil()
+    end
+end)
+
+
+windower.register_event('incoming text', function(...)
+    update_gil()
+end)
+
+function initialize()
+    hideKey = settings.HideKey
+    local xRes = windower.get_windower_settings().ui_x_res
+    local yRes = windower.get_windower_settings().ui_y_res
+    update_gil()
+    gil_image:pos(xRes + settings.GilText.pos.x + 1 ,
+      yRes + settings.GilText.pos.y - (settings.GilImage.size.height/6))
+    show()
+end
+
+function update_gil()
+  local gil = windower.ffxi.get_items().gil
+  gil_text:text(''..comma_value(gil))
+end
+
+function show()
+    gil_text:show()
+    gil_image:show()
+end
+
+function hide()
+    gil_text:hide()
+    gil_image:hide()
+end
+
+function comma_value(amount)
+  local formatted = amount
+  while true do
+    formatted, k = string.gsub(formatted, "^(-?%d+)(%d%d%d)", '%1,%2')
+    if (k==0) then
+      break
+    end
+  end
+  return formatted
+end
+
+windower.register_event('status change', function(new_status_id)
+    if (new_status_id == 4) and (is_hidden_by_key == false) then
+        is_hidden_by_cutscene = true
+        hide()
+    elseif (new_status_id ~= 4) and (is_hidden_by_key == false) then
+        is_hidden_by_cutscene = false
+        show()
+    end
+end)
+
+windower.register_event('keyboard', function(dik, flags, blocked)
```

Comment:
`flags` is an odd name for this. This parameter indicates only if the button was pressed down or released. So a name like `down` would be more appropriate, I guess.

Reply:
There's a fourth parameter for the `keyboard` event. I think it's `function(dik, down, flags, blocked)`. There must have been an example or an early addon where someone forgot the `down` parameter, because I remember copy/pasting the same a few times.

It won't cause you trouble unless you check the `blocked` parameter expecting to see if another addon has handled the event. Currently `blocked` is being passed the `flags` data, which will always short-circuit to `true`.

Reply:
You're right, and they already adjusted it accordingly. The confusion is actually kind warranted, because whether the key is pressed or released is normally part of the flags, but since it was so commonly used (in comparison to the other flags) we decided to extract it into its own parameter on the LuaCore side.